lib/code_fund/sponsorships/sponsorships.ex

```diff
@@ -94,41 +95,53 @@ defmodule CodeFund.Sponsorships do
     |> Repo.preload([:property, :campaign, :creative, :user])
   end
 
-  def get_sponsorship_for_property(%Property{} = property) do
-    campaign =
-      Repo.one(
-        from(
-          c in Campaign,
-          join: s in assoc(c, :sponsorships),
-          join: b in assoc(c, :budgeted_campaign),
-          where: c.status == 2,
-          where: s.property_id == ^property.id,
-          where: b.day_remain > 0,
-          where: b.month_remain > 0,
-          where: b.total_remain > 0,
-          order_by: [desc: c.bid_amount],
-          limit: 1
-        )
-      )
-
-    if is_nil(campaign) do
-      Property.changeset(property, %{sponsorship_id: nil}) |> Repo.update()
-      nil
-    else
-      sponsorship =
-        Repo.one(
-          from(
-            s in Sponsorship,
-            where: s.property_id == ^property.id,
-            where: s.campaign_id == ^campaign.id,
-            limit: 1
-          )
-        )
-        |> Repo.preload([:campaign, :property, :creative, :user])
+  def get_and_update_sponsorship_for_property(
+        %Property{} = property,
+        campaign_order \\ :bid_amount
+      ) do
+    case campaign_order do
```

Comment:
Would we ever want this case? It seems that this should always return the random selection from the highest matching bidders. Thoughts?

Reply:
Idk, I didn't want to kill existing functionality that I wasn't sure if we needed as this function is called from more than one place.

Reply:
I believe it should always be chosen randomly from the top matching bids.